Tasks/FtpUploadV2/ftpuploadtask.ts

```diff
@@ -23,142 +20,322 @@ export class FtpOptions {
+        options.serverEndpointUrl = url.parse(tl.getInput("serverUrl", true));
+        options.username = tl.getInput("username", true);
+        options.password = tl.getInput("password", true);
     }
 
     // other standard options
-    options.rootFolder = tl.getPathInput('rootFolder', true);
-    options.filePatterns = tl.getDelimitedInput('filePatterns', '\n', true);
-    options.remotePath = tl.getInput('remotePath', true).trim();
+    options.rootFolder = tl.getPathInput("rootFolder", true);
+    options.filePatterns = tl.getDelimitedInput("filePatterns", "\n", true);
+    options.remotePath = tl
+        .getInput("remotePath", true)
+        .trim()
+        .replace(/\\/gi, "/"); // use forward slashes always
 
     // advanced options
-    options.clean = tl.getBoolInput('clean', true);
-    options.cleanContents = tl.getBoolInput('cleanContents', false);
-    options.overwrite = tl.getBoolInput('overwrite', true);
-    options.preservePaths = tl.getBoolInput('preservePaths', true);
-    options.trustSSL = tl.getBoolInput('trustSSL', true);
+    options.clean = tl.getBoolInput("clean", true);
+    options.cleanContents = tl.getBoolInput("cleanContents", false);
+    options.preservePaths = tl.getBoolInput("preservePaths", true);
+    options.trustSSL = tl.getBoolInput("trustSSL", true);
 
     return options;
 }
 
-function doWork() {
-    tl.setResourcePath(path.join( __dirname, 'task.json'));
+function getAccessOption(options: FtpOptions): ftp.AccessOptions {
+    let secure: boolean =
+        options.serverEndpointUrl.protocol.toLowerCase() == "ftps:" ? true : false;
+    let secureOptions: any = { rejectUnauthorized: !options.trustSSL };
 
-    let ftpOptions: FtpOptions = getFtpOptions();
-    if (!ftpOptions.serverEndpointUrl.protocol) {
-        tl.setResult(tl.TaskResult.Failed, tl.loc('FTPNoProtocolSpecified'));
-    }
-    if (!ftpOptions.serverEndpointUrl.hostname) {
-        tl.setResult(tl.TaskResult.Failed, tl.loc('FTPNoHostSpecified'));
+    let hostName: string = options.serverEndpointUrl.hostname;
+    let portStr: string = options.serverEndpointUrl.port;
+    let port: number = 21;
+    if (portStr) {
+        // port not explicitly specifed, use default
+        port = parseInt(portStr);
     }
 
-    let ftpClient: any = new Client();
-    let ftpHelper: ftputils.FtpHelper = new ftputils.FtpHelper(ftpOptions, ftpClient);
+    console.log(tl.loc("ConnectPort", hostName, port));
 
-    let files: string[] = ftputils.findFiles(ftpOptions);
-    tl.debug('number of files to upload: ' + files.length);
-    tl.debug('files to upload: ' + JSON.stringify(files));
+    return {
+        host: hostName,
+        port: port,
+        user: options.username,
+        password: options.password,
+        secure: secure,
+        secureOptions: secureOptions
+    };
+}
 
-    let uploadSuccessful: boolean = false;
+async function getFtpClient(options: FtpOptions): Promise<ftp.Client> {
+    let ftpClient = new ftp.Client();
```

Comment:
`const`?

Reply:
No, may need to reassign it to other client instance if we got disconnected.